### DISCUSSION TITLE
Web worker and typescript, what's the required config to have a web worker setup ?

### DISCUSSION THREAD

**Discussion by Mica4DEV** [alex8088/electron-vite#445] [Q&A]
Hi everyone,

I've searched accross the web but except for webpack I can't seem to find any solution for an usage with Vite and:
- Typescript
- Web workers

Every attempts to load a worker from the rendered process vue file fails.

The last attempt looks as follows:

App.vue:
`
const worker = new Worker(new URL('./workers/worker?modulePath', import.meta.url));
worker.postMessage('message from main thread');
worker.onmessage = (ev: MessageEvent) => {
  console.log('do-something-in-renderer-thread', 'I received as a response from the worker', ev.data);
};
`

src/workers/worker.ts
`
self.onmessage = (e: MessageEvent<string>) => {
  self.postMessage('response from worker for a message ' + e.data + ' :' + new Date().toISOString());
};
`

But if fails indicating that it tried to load an url that is in fact the rendered vue html app (probably taking the 404 not found behaviour of the app).

It's like it did not build any web worker file so it takes the app default page instead.

If any one has an idea on how to setup things for that it'd be very appreciated.

Thanks,

**Comment by alex8088** [comment]
`?modulePath` only applies to main process and node workers, Learn more about [web worker](https://vitejs.dev/guide/features.html#web-workers) in Vite.
